pkg/webui/console/containers/fetch-select/index.js

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { connect as storeConnect } from 'react-redux'
+
+import PropTypes from '../../../lib/prop-types'
+import Field from '../../../components/form/field'
+import Select from '../../../components/select'
+
+const formatOptions = formats =>
+  Object.keys(formats).map(key => ({ value: key, label: formats[key] }))
+
+export default function({
+  optionsSelector,
+  errorSelector,
+  fetchingSelector,
+  fetchOptions,
+  warningMessage,
+  titleMessage,
+  optionsFormatter = formatOptions,
+}) {
+  @storeConnect(
+    function(state) {
+      return {
+        options: optionsSelector(state),
+        error: errorSelector(state),
+        fetching: fetchingSelector(state),
+      }
+    },
+    { fetchOptions },
+  )
+  class FetchSelect extends React.PureComponent {
+    static propTypes = {
+      autoFocus: PropTypes.bool,
+      error: PropTypes.error,
+      fetchOptions: PropTypes.func.isRequired,
+      fetching: PropTypes.bool,
+      menuPlacement: PropTypes.oneOf(['top', 'bottom', 'auto']),
+      name: PropTypes.string.isRequired,
+      options: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.arrayOf(PropTypes.shape({}))]),
```

Comment:
I believe thats the shape of options:
```suggestion
      options: PropTypes.arrayOf(PropTypes.shape({value: PropTypes.string, label: PropTypes.message})),
```